src/test/java/nl/openweb/hippo/mock/MockComponentManagerTest.java

```diff
@@ -0,0 +1,146 @@
+
+        Assert.assertEquals(servletConfig, componentManager.getServletConfig());
+        Assert.assertEquals(context, componentManager.getServletContext());
+    }
+
+    @Test
+    public void getComponentsOfTypeTest() {
+        String component1 = "component1";
+        String component2 = "component2";
+        componentManager.addComponent("test1", component1);
+        componentManager.addComponent("test2", component2);
+
+        Map<String, String> componentsOfType = componentManager.getComponentsOfType(String.class);
+        Assert.assertEquals(2, componentsOfType.size());
+    }
+
+    @Test
+    public void getComponentsOfTypeTestWithModuleName() {
+        String component1 = "component1";
+        String component2 = "component2";
+        componentManager.addComponent("test1", component1);
+        componentManager.addComponent("test2", component2);
+
+        Map<String, String> componentsOfType = componentManager.getComponentsOfType(String.class, "foo");
+        Assert.assertEquals(2, componentsOfType.size());
+    }
+
+    @Test
+    public void getComponentTest() {
+        String component1 = "component1";
+        String component2 = "component2";
+        componentManager.addComponent("test1", component1);
+        componentManager.addComponent("test2", component2);
+
+        String result1 = componentManager.getComponent("test1");
+        String result2 = componentManager.getComponent("component1");
+        Assert.assertEquals(component1, result1);
+        Assert.assertNull(result2);
+    }
+
+    @Test
+    public void getComponentTestWithModuleName() {
+        String component1 = "component1";
+        String component2 = "component2";
+        componentManager.addComponent("test1", component1);
+        componentManager.addComponent("test2", component2);
+
+        String result1 = componentManager.getComponent("test1", "foo");
+        String result2 = componentManager.getComponent("component1", "foo");
+        Assert.assertEquals(component1, result1);
+        Assert.assertNull(result2);
+    }
+
+    @Test
+    public void getComponentByClassTest() {
+        String component1 = "component1";
+        componentManager.addComponent("test1", component1);
+
+        String result1 = componentManager.getComponent(String.class);
+        Assert.assertEquals(component1, result1);
+        Double result2 = componentManager.getComponent(Double.class);
+        Assert.assertNull(result2);
+    }
+
+    @Test(expected = ComponentsException.class)
+    public void getComponentByClassTestWithMultipleComponentsOfSameType() {
+        String component1 = "component1";
+        Integer component2 = 2;
+        Integer component3 = 3;
+        componentManager.addComponent("test1", component1);
+        componentManager.addComponent("test2", component2);
+        componentManager.addComponent("test3", component3);
+
+        componentManager.getComponent(Integer.class);
+        Assert.fail();
```

Comment:
With the @Test (expected=..) when it didn't throw the error the test will fail.
Naming of the test isn't telling it's testing an Exception